File: parameterized_tests/csharp/ParameterizedTests/DomainTests/PremiumFactorsWithinRangeTest.cs
using Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using NUnit.Framework;

namespace DomainTests
{
    public class PremiumFactorsWithinRangeTest
    {
        [Datapoints]
        public Gender[] genders
                = new Gender[]{Gender.Female,
                Gender.Male, Gender.Unknown};

        [Datapoints]
        public int[] ages
                = new int[]{17, 18, 19, 23, 24, 25,
            59, 60, 61, 100, 101};

        [Theory]
        public void PremiumFactorsAreBetween0_5and2_0(Gender gender, int age)
        {
            Assume.That(age, Is.GreaterThanOrEqualTo(18));
            Assume.That(age, Is.LessThanOrEqualTo(100));
            Assume.That(gender == Gender.Female || gender == Gender.Male);
            double premiumFactor
                    = new PremiumRuleEngine().GetPremiumFactor(age, gender);
            Assert.That(premiumFactor, Is.InRange(0.5, 2.0));
        }
    }
}

File: parameterized_tests/java/ParameterizedTests/src/test/java/premiums/domain/PremiumFactorsWithinRangeTest.java
package premiums.domain;

import org.junit.experimental.theories.DataPoints;
import org.junit.experimental.theories.Theories;
import org.junit.experimental.theories.Theory;
import org.junit.runner.RunWith;

import static org.hamcrest.Matchers.*;
import static org.junit.Assert.assertThat;
import static org.junit.Assume.assumeThat;


@RunWith(Theories.class)
public class PremiumFactorsWithinRangeTest {

    @DataPoints
    public static Gender genders[]
            = new Gender[]{Gender.FEMALE,
            Gender.MALE, Gender.UNKNOWN};

    @DataPoints
    public static int ages[]
            = new int[]{17, 18, 19, 23, 24, 25,
            59, 60, 61, 100, 101};

    @Theory
    public void premiumFactorsAreBetween0_5and2_0(Gender gender, int age) {
        assumeThat(age, greaterThanOrEqualTo(18));
        assumeThat(age, lessThanOrEqualTo(100));
        assumeThat(gender, isOneOf(Gender.FEMALE, Gender.MALE));

        double premiumFactor
                = new PremiumRuleEngine().getPremiumFactor(age, gender);
        assertThat(premiumFactor,
                is(both(greaterThan(0.5)).and(lessThan(2.0))));
    }
}
